### DISCUSSION TITLE
can you use GRBL for laser on a Arduino Mega 2560

### DISCUSSION THREAD

**Discussion by acourtjester** [arkypita/LaserGRBL#1802] [General]
I am trying to use a Arduino Mega 2560 to run a laser on my DIY CNC table.  I have searched and found a lot of info but no clear answer.  I have flashed the Mega and Laser GRBL will see the com port for the Mega but no communications with it.  Basically the same with Lightburn.  Lights flash when looking at the Mega nothing else happens.  Is there a different load for the Mega to work with GRBL??  I have and will use a Uno as it seems for communicate with the software, I did make the changes to the GRBL load and I am about to wire it up for testing
Thanks Tom

**Comment by gmmanonymus111** [comment]
Hey, sorry for the delayed reply,
AFAIK, it is posible to use a Mega for a CNC, as I've used it myself. Keep in mind, that you need a different firmware, as the pinout is different from the Uno. I'd recommend checking out [grbl-mega-5x](https://github.com/fra589/grbl-Mega-5X), as this was the firmware I used, and it was working great.
For debugging the firmware (or, to check if it's flashed correctly), you can use Arduino IDE's terminal, or [PuTTY](https://www.putty.org/). Just set the COM port, and the BAUD rate, hit connect, then type `?` to get a status report, or `$$` to list your machine's current settings.
Could you attach a picture of your machine/setup?

**Reply by acourtjester** [reply]
thanks I will look that over  Here ian image showing 2 of table I have built (9 so far) this are used for plasma, routing, drag knife, and metal scribing. I built both 4X4 and 4X8 table from scratch.
![2 tables 4x4 and new 4x8](https://user-images.githubusercontent.com/3810543/162484042-2b1b5134-bd8c-4784-8a45-8eddb17de98e.JPG)
r

**Comment by acourtjester** [comment]
Thanks Alex  big help with this

**Reply by acourtjester** [reply]
Thanks again I did have an upload problem but after watching and following this video it finally worked
https://www.youtube.com/watch?v=jNKa2-UwKgo

**Reply by gmmanonymus111** [reply]
I'm glad it worked out! Happy tinkering ^^
